### PR TITLE
feat: map-block parameter for disable auto zoom

### PR DESCRIPTION
**Pull request by Lakate**

### CODE REVIEW

**Review by gorgeousvlad — APPROVED** [review]

**Comment by Lakate** [comment]
Some changes:
1) delete center from params - we always calculate it in the code. We show the Yandex map after calculating the center
2) zoom - if you specify it in the parameters, we use it, if not, then we calculate it in the code

File: src/components/Map/GoogleMap.tsx [hunk 1]
```diff
@@ -10,21 +10,29 @@ import {getMapHeight} from './helpers';
...
 const b = block('map');
 
-function getScriptSrc(apiKey: string, scriptSrc: string, address: string, lang: 'ru' | 'en') {
-    return `${scriptSrc}?key=${apiKey}&language=${lang}&q=${encodeURI(address)}`;
+function getScriptSrc(
+    apiKey: string,
+    scriptSrc: string,
+    address: string,
+    lang: 'ru' | 'en',
+    zoom?: number,
```
Comment (gorgeousvlad) [inline src/components/Map/GoogleMap.tsx:18]: to many serial params in function let's rewrite to object

File: src/components/Map/GoogleMap.tsx [hunk 2]
```diff
@@ -10,21 +10,29 @@ import {getMapHeight} from './helpers';
 
 const b = block('map');
 
-function getScriptSrc(apiKey: string, scriptSrc: string, address: string, lang: 'ru' | 'en') {
-    return `${scriptSrc}?key=${apiKey}&language=${lang}&q=${encodeURI(address)}`;
+function getScriptSrc(
+    apiKey: string,
+    scriptSrc: string,
+    address: string,
+    lang: 'ru' | 'en',
```
Comment (gorgeousvlad) [inline src/components/Map/GoogleMap.tsx:17]: do we already had enum for Lang?

File: src/components/Map/Map.scss [hunk 3]
```diff
@@ -10,9 +10,14 @@ $block: '.#{$ns}map';
     overflow: hidden;
     display: flex;
 
+    &_visible_no {
```
Comment (gorgeousvlad) [inline src/components/Map/Map.scss:13]: can we replace `visible_no` to `hidden`?

File: src/components/Map/YMap/YMap.ts [hunk 4]
```diff
@@ -90,8 +88,9 @@ export class YMap {
         );
     }
 
-    private recalcZoomAndCenter() {
+    private recalcZoomAndCenter(props: PlacemarksProps) {
         const coordsLength = this.coords.length;
+        const {zoom} = props;
```
Comment (gorgeousvlad) [inline src/components/Map/YMap/YMap.ts:93]: const {zoom=0} = props;

File: src/components/Map/YMap/YMap.ts [hunk 5]
```diff
@@ -105,17 +104,27 @@ export class YMap {
...
-            {margin: DEFAULT_MAP_CONTROL_BUTTON_HEIGHT},
-        );
+        let newMapParams = {
+            zoom: zoom || 0,
+            center: [],
+        };
+
+        if (zoom) {
+            // compute only the center
+            newMapParams.center = window.ymaps.util.bounds.getCenter([leftBottom, rightTop]);
```
Comment (gorgeousvlad) [inline src/components/Map/YMap/YMap.ts:114]: can we get the situation when window.ymaps or window.ymaps.util or window.ymaps.util.bounds is undefined?

File: src/components/Map/YMap/YandexMap.tsx [hunk 6]
```diff
@@ -16,9 +16,10 @@ import {getMapHeight} from '../helpers';
 const b = block('map');
 const DEFAULT_CONTAINER_ID = 'ymap';
 const DEFAULT_ZOOM = 9;
+const DEFAULT_CENTER = [55.755864, 37.617698]; // will be calculated later
```
Comment (gorgeousvlad) [inline src/components/Map/YMap/YandexMap.tsx:19]: did't understand meaning of the comment

File: src/components/Map/YMap/YandexMap.tsx [hunk 7]
```diff
@@ -96,7 +101,14 @@ const YandexMap: React.FC<YMapProps> = (props) => {
             handler={onTryAgain}
             className={b('wrapper')}
         >
-            <div id={containerId} className={b()} ref={ref} style={{height}}>
+            <div className={b('wrapper')}>
+                {/* visible - to show the map after calculating the center */}
+                <div
+                    id={containerId}
+                    className={b({visible: ready ? 'yes' : 'no'})}
```
Comment (gorgeousvlad) [inline src/components/Map/YMap/YandexMap.tsx:108]: `className={b({hidden: !Boolean(ready)})}`

**Review by gorgeousvlad — CHANGES_REQUESTED** [review]

**Review by gorgeousvlad — APPROVED** [review]

**Pull request merged**